tls/s2n_connection.h

```diff
@@ -326,3 +334,5 @@ int s2n_connection_set_client_auth_type(struct s2n_connection *conn, s2n_cert_au
 int s2n_connection_get_client_auth_type(struct s2n_connection *conn, s2n_cert_auth_type *client_cert_auth_type);
 int s2n_connection_get_client_cert_chain(struct s2n_connection *conn, uint8_t **der_cert_chain_out, uint32_t *cert_chain_len);
 uint8_t s2n_connection_get_protocol_version(const struct s2n_connection *conn);
+/* IANA_ID = 0 is RESERVED and is used to represent sending a list of empty keyshares */
+int s2n_connection_set_keyshare_by_group_for_testing(struct s2n_connection *conn, uint16_t iana_id);
```

Comment:
I think it's better not to overload separate meaning onto values in the IANA registry. Especially if it's easily avoidable. RESERVED does not necessarily mean "will never be used" and it may cause confusion or problems later when folks forget about this implied behavior.

Can we instead signal "send empty key shares" outside of the IANA argument? For s2nc, we could encode send no key shares as something that's clear a non numerical string. Something like: `./bin/s2nc -k 'NONE'`. And add a new function `s2n_connection_set_no_keyshares()` or something similar?

Reply:
That was my initial change, I was requested to not introduce more APIs and have a single API to represent all cases.

Reply:
We could use the curve_name instead of the iana_id, this way we won't have to use the RESERVED. `x25519`, `secp384r1` and `secp256r1` to represent the three supported curves and `empty` to send an empty keyshare.